README.markdown

```diff
@@ -449,6 +449,22 @@ If you want to stop processing jobs, but want to leave the worker running
 (for example, to temporarily alleviate load), use `USR2` to stop processing,
 then `CONT` to start it again.
 
+#### Signals on Heroku
+
+When shutting down processes, Heroku sends every process a TERM signal at the
+same time. By default this causes an immediate shutdown of any running job
+leading to frequent `Resque::TermException` errors.  For short running jobs, a simple
+solution is to give a small amount of time for the job to finish
+before killing it.
+
+To accomplish this set the following environment variables:
+
+* `RESQUE_PRE_SHUTDOWN_TIMEOUT` - The time between the parent receiving a shutdown signal (TERM by default) and it sending that signal on to the child process. Designed to give the child process
+time to complete before being forced to die.
+
+* `RESQUE_SHUTDOWN_SIGNAL` - The signal sent to the child process
```

Comment:
We removed the config and ensured it works as it should on heroku. In order for it to work `TERM_CHILD` needs to be set. We updated the readme to reflect that.

When heroku sends a TERM to the parent process, it waits for `RESQUE_PRE_SHUTDOWN_TIMEOUT`, then sends TERM to the child, and waits `RESQUE_TERM_TIMEOUT` for it to handle `Resque::TermException`. This works as long as both timeouts combines is less than the heroku timeout before it sends KILL (currently 30 seconds).